crates/sui-rosetta/src/operations.rs

```diff
@@ -662,17 +739,26 @@ impl Operations {
         }
 
         // Extract coin change operations from balance changes
-        let coin_change_operations = Self::process_balance_change(
+        let mut coin_change_operations = Self::process_balance_change(
             gas_owner,
             gas_used,
             balance_changes,
             status,
-            accounted_balances,
+            accounted_balances.clone(),
+        );
+
+        let single_gascoin_transfer = Self::process_single_gascoin_transfer(
+            &mut coin_change_operations,
+            tx.data.transaction().clone(),
+            tx.data.gas_data().owner,
+            gas_owner,
+            gas_used,
         );
 
```

Comment:
Every transaction that happens on-chain will pass from here. We need to be extra careful that we do not mess up the balance-changes. Does it make sense to add a sanity check that operations balance changes match the tx balance changes after editing the operations with gas-coins transfer?
@patrickkuo , @nonvis wdyt?

Reply:
I don't think a sanity check is required. First, the `balance-changes` is edited only when GasCoin is referenced for transferObjects, which occurs infrequently. And if they do occur, only two types of operations are used: `BalanceChange` and `Gas`. `process_gascoin_transfer` function processes these two types of operations.
So in terms of the number of operations stored before/after `process_gascoin_transfer`, they would be the same.